libr/debug/debug.c

```diff
@@ -1673,3 +1673,19 @@ R_API ut64 r_debug_get_baddr(RDebug *dbg, const char *file) {
 	return 0LL;
 #endif
 }
+
+R_API void r_debug_bp_rebase(RDebug *dbg, ut64 baddr) {
+	RBreakpointItem *bp;
+	RListIter *iter;
+	r_list_foreach (dbg->bp->bps, iter, bp) {
+		bp->addr += baddr;
```

Comment:
this seems like a bad idea. maybe the breakpoints should store more information in order to hold the reference to the associated map or bin, as long as this can be very different between versions it can be tricky.

this code can:

* overflow the addr value
* if called twice will result on invalid results

What about holding bp->oaddr and make bp->addr = bp->oaddr + baddr ? we will still need a way to reload the breakpoints to the right places

Reply:
That seems a better way to do it, yes. If fixed the issue when calling this twice would invalidate. When unrebasing I update the orig_addr of all of them but it should be the same for those that were rebased, and the correct one for those bps added during the debug